pkg/provider/cloud/kubevirt/manifests/manifests_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifests
+
+import (
+	"embed"
+	"path"
+	"reflect"
+	"testing"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	kvinstancetypev1alpha1 "kubevirt.io/api/instancetype/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntimefakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var (
+	fakeclient ctrlruntimeclient.Client
+	//go:embed test/presetOk
+	testPresetFS     embed.FS
+	testPresetFolder = path.Join("test", "presetOk")
+
+	//go:embed test/presetOkKo
+	tesOkKoPresetFS      embed.FS
+	testOkKoPresetFolder = path.Join("test", "presetOkKo")
```

Comment:
as discussed let's remove `presetOkKo` and keep only one directory for all preset fixtures. The invalid one you can prefix with `invalid-` and later check in the code. Additionaly please add invalid resource for preferences and instancetypes as well.

Reply:
done